### PR TITLE
fix(security): vulnerabilities + remove coveralls

### PR DESCRIPTION
**Pull request by benjlevesque**
## Changes
This PR upgrades Lerna from 3 to 6, there is no breaking change and the new version is actually pretty nice. We'll need to confirm publish still works. NB: Lerna was the cause of _many_ vulnerabilities. 

It also drops `coveralls`  and the `publish-coverage` step from the CI. I don't think code coverage is worth having critical vulnerabilities, and there's no fix available (the library hasn't been maintained for 3 years)

## Resolutions
Before: 
```
$ yarn audit
yarn audit v1.22.19
[...]
58 vulnerabilities found - Packages audited: 1699
Severity: 19 Moderate | 32 High | 7 Critical
Done in 2.92s.
```

After:
```
$ yarn audit
yarn audit v1.22.19
0 vulnerabilities found - Packages audited: 1611
Done in 2.58s.
```
_NB: this is optimistic, the OpenZeppelin vulnerabilities will not be removed by this PR_

### Critical
https://github.com/RequestNetwork/requestNetwork/security/dependabot/207
https://github.com/RequestNetwork/requestNetwork/security/dependabot/177
https://github.com/RequestNetwork/requestNetwork/security/dependabot/137
https://github.com/RequestNetwork/requestNetwork/security/dependabot/123
https://github.com/RequestNetwork/requestNetwork/security/dependabot/51
https://github.com/RequestNetwork/requestNetwork/security/dependabot/14

### High 
https://github.com/RequestNetwork/requestNetwork/security/dependabot/182
https://github.com/RequestNetwork/requestNetwork/security/dependabot/172
https://github.com/RequestNetwork/requestNetwork/security/dependabot/165
https://github.com/RequestNetwork/requestNetwork/security/dependabot/124
https://github.com/RequestNetwork/requestNetwork/security/dependabot/120
https://github.com/RequestNetwork/requestNetwork/security/dependabot/113
https://github.com/RequestNetwork/requestNetwork/security/dependabot/101
https://github.com/RequestNetwork/requestNetwork/security/dependabot/80
https://github.com/RequestNetwork/requestNetwork/security/dependabot/75
https://github.com/RequestNetwork/requestNetwork/security/dependabot/45

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,6 @@
     "lint-staged": "lint-staged",
     "lerna": "lerna",
     "packageJsonLint": "npmPkgJsonLint ./packages",
-    "publish-coverage": "mkdir -p ./coverage && node scripts/collect.js && nyc merge ./coverage ./coverage/coverage.json && node scripts/remap.js && cat coverage/lcov.info | coveralls",
```
Comment (alexandre-abrioux) [inline package.json:26]: Should we keep this command for reference and pipe to nothing:
```bash
| echo "TODO: send to coveralls with coverallsapp/coverage-reporter"
```
; or should we remove the `scripts` folder altogether?

**Review by alexandre-abrioux — APPROVED** [review]
🤩 amazing

**Review by kevindavee — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (benjlevesque) [reply package.json:26]: and it enables to remove 1 dependency 🤩

Comment (benjlevesque) [reply package.json:26]: we can also remove nyc

Comment (benjlevesque) [reply package.json:26]: thanks, I was looking for that one!

**Pull request merged**